pkg/istio-agent/xds_proxy.go

```diff
@@ -708,3 +722,108 @@ func sendWithTimeout(ctx context.Context, sendFunc func(errorChan chan error)) e
 		return err
 	}
 }
+
+// tapRequest() sends "req" to Istiod, and returns a matching response, or `nil` on timeout.
+// Requests are serialized -- only one may be in-flight at a time.
+func (p *XdsProxy) tapRequest(req *discovery.DiscoveryRequest, timeout time.Duration) (*discovery.DiscoveryResponse, error) {
+	if p.connected == nil {
+		return nil, fmt.Errorf("proxy not connected to Istiod")
+	}
+
+	defer p.tapMutex.Unlock()
```

Comment:
nit: I suppose this works too but it seems more common to add this immediately after the .Lock()